appservice/src/deploy/waitForDeploymentToComplete.ts

```diff
@@ -24,7 +29,7 @@ export async function waitForDeploymentToComplete(client: SiteClient, expectedId
     const kuduClient: KuduClient = await client.getKuduClient();
 
     while (!token || !token.isCancellationRequested) {
-        [deployment, permanentId, initialStartTime] = await tryGetLatestDeployment(kuduClient, permanentId, initialStartTime, expectedId);
+        [deployment, permanentId, initialStartTime] = await tryGetLatestDeployment(context, kuduClient, permanentId, initialStartTime, expectedId);
```

Comment:
Would it work to just use the `await retry()` logic on this call instead of doing it separately for each case in `tryGetLatestDeployment`?

Reply:
Doing that has a slight risk of retrying something that shouldn't be retried (like a null ref exception). I went back and forth on whether the cleaner code is worth the slight risk, but Imma leave it as-is for now. Ultimately I want to check telemetry after this is released and refactor anyways:
1. If it's not helping, remove this logic
1. If it is helping, add generic retry logic for _all_ kudu calls